c7n/resources/ebs.py

```diff
@@ -454,6 +454,86 @@ def process(self, resources, event=None):
         return self.get_matching_aliases(resources)
 
 
+@Snapshot.filter_registry.register('invalid-kms')
+class KmsKeyActiveSnapshot(FilterKmsInvalid):
+    """
+    Filter to ignore snapshots that have an active KMS Key.
+
+    This filter is 'true' by default.
+
+    :example:
+
+    implicit with no parameters, 'true' by default
+
+    .. code-block:: yaml
+
+            policies:
+              - name: delete-snapshots-with-missing-keys
+                resource: ebs-snapshot
+                filters:
+                  - invalid-kms
+
+    :example:
+
+    explicit with parameter
+
+    .. code-block:: yaml
+
+            policies:
+              - name: delete-snapshots
+                resource: ebs-snapshot
+                filters:
+                  - type: invalid-kms
+                    value: false
+
+    """
+
+    schema = type_schema('invalid-kms', value={'type': 'boolean'})
+
+    def get_permissions(self):
```

Comment:
this can be implemented on the base class, and the permission should be for 'kms' instead of 'ec2' resource manager.